src/qililab/instruments/qblox/qblox_module.py

```diff
@@ -137,168 +121,16 @@ def module_type(self):
-            waveform_pair = waveforms.find_pair_by_name(pulse_event.pulse.label())
-            wait_time = timeline[i + 1].start_time - pulse_event.start_time if (i < (len(timeline) - 1)) else 4
-            phase = int((pulse_event.pulse.phase % (2 * np.pi)) * 1e9 / (2 * np.pi))
-            gain = int(np.abs(pulse_event.pulse.amplitude) * AWG_MAX_GAIN)  # np.abs() needed for negative pulses
-            bin_loop.append_component(SetAwgGain(gain_0=gain, gain_1=gain))
-            bin_loop.append_component(SetPh(phase=phase))
-            bin_loop.append_component(
-                Play(
-                    waveform_0=waveform_pair.waveform_i.index,
-                    waveform_1=waveform_pair.waveform_q.index,
-                    wait_time=int(wait_time),
-                )
-            )
-        self._append_acquire_instruction(
-            loop=bin_loop, bin_index=bin_loop.counter_register, sequencer_id=sequencer, weight_regs=weight_registers
-        )
-        if self.repetition_duration is not None:
-            wait_time = self.repetition_duration - bin_loop.duration_iter
-            if wait_time > self._MIN_WAIT_TIME:
-                bin_loop.append_component(long_wait(wait_time=wait_time))
-
-        logger.info("Q1ASM program: \n %s", repr(program))  # pylint: disable=protected-access
-        return program
-
-    def _init_weights_registers(self, registers: tuple[Register, Register], values: tuple[int, int], program: Program):
-        """Initialize the weights `registers` to the `values` specified and place the required instructions in the
-        setup block of the `program`."""
-
-    def _generate_acquisitions(self) -> Acquisitions:
-        """Generate Acquisitions object, currently containing a single acquisition named "default", with num_bins = 1
-        and index = 0.
-
-        Returns:
-            Acquisitions: Acquisitions object.
-        """
-        # FIXME: is it really necessary to generate acquisitions for a QCM??
-        acquisitions = Acquisitions()
-        acquisitions.add(name="default", num_bins=self.num_bins, index=0)
-        return acquisitions
-
-    @abstractmethod
-    def _generate_weights(self, sequencer: AWGQbloxSequencer) -> Weights:
-        """Generate acquisition weights.
-
-        Returns:
-            dict: Acquisition weights.
-        """
-
-    @abstractmethod
-    def _append_acquire_instruction(
-        self, loop: Loop, bin_index: Register | int, sequencer_id: int, weight_regs: tuple[Register, Register]
-    ):
-        """Append an acquire instruction to the loop."""
-
     def start_sequencer(self, port: str):
         """Start sequencer and execute the uploaded instructions."""
         sequencers = self.get_sequencers_from_chip_port_id(chip_port_id=port)
         for sequencer in sequencers:
             if sequencer.identifier in self.sequences:
+                # TODO: skip sequencer here if it is not in cache, instead of poping the sequence at upload
```

Comment:
this is because I wanted to handle cache in the compiler, and sequences here in qblox_module (do decouple compiler and sequencer as much as possible). But in the end we agreed (in the code discussion before Christmas) to handle both at compiler